ci-operator/step-registry/cucushift/installer/check/reboot-nodes/cucushift-installer-check-reboot-nodes-commands.sh

```diff
@@ -52,32 +353,106 @@ function reboot_cluster() {
         node_ip_array[${node_name}]=${node_ip}
     done
 
-    for node_name in ${node_list}; do
-        node_ip=${node_ip_array[${node_name}]}
-        echo "$(date -u --rfc-3339=seconds) - rebooting node ${node_name}, node ip is ${node_ip}"
-        reboot_node "${node_ip}"
-    done
-
-    total_nodes_count=$(echo ${node_list} | awk '{print NF}')
-    try=0
-    max_try=30
-    while [[ ${try} -lt ${max_try} ]]; do
-	if [[ $(oc get node --no-headers | grep -c 'Ready') -eq ${total_nodes_count} ]]; then
-            echo "$(date -u --rfc-3339=seconds) - cluster boot up, get ready"
-            break
-        fi
-        echo "$(date -u --rfc-3339=seconds) - wait for node boot up"
-        sleep 60
-        try=$(( try + 1 ))
-    done
+    echo "$(date -u +"%Y-%m-%dT%H:%M:%SZ") - Using reboot strategy: ${REBOOT_STRATEGY}"
+    
+    if [[ "${REBOOT_STRATEGY}" == "BATCH_STOP_START" && "${CLUSTER_TYPE}" == "aws" ]]; then
+        # Batch stop all nodes first
+        echo "$(date -u +"%Y-%m-%dT%H:%M:%SZ") - Batch stopping all nodes..."
+        for node_name in ${node_list}; do
+            node_ip=${node_ip_array[${node_name}]}
+            echo "$(date -u +"%Y-%m-%dT%H:%M:%SZ") - Stopping node ${node_name} (${node_ip})"
+            if ! reboot_node_stop_only "${node_ip}" "${node_name}"; then
+                echo "$(date -u +"%Y-%m-%dT%H:%M:%SZ") - ERROR: Failed to stop node ${node_name}, continuing with other nodes..."
+            fi
+        done
+        
+        # Wait fixed time for all nodes to stop
+        echo "$(date -u +"%Y-%m-%dT%H:%M:%SZ") - Waiting 2 minutes for all nodes to stop completely..."
+        sleep 120
```

Comment:
can we use `aws` command to check/wait for all nodes to be stopped, rather than using a fixed sleep time?